data_steward/tools/load_vocab.py

```diff
@@ -0,0 +1,270 @@
+"""
+ Load an Athena vocabulary bundle located in a GCS bucket into a BQ dataset
+
+"""
+import argparse
+import datetime
+import logging
+from typing import List, Iterable, Union
+
+import jinja2
+from google.cloud import storage
+from google.cloud.bigquery import Client, Dataset, SchemaField, LoadJob, LoadJobConfig, QueryJob, \
+    QueryJobConfig, Table
+
+from common import VOCABULARY_TABLES
+from utils.sandbox import get_sandbox_dataset_id
+from utils import bq
+
+LOGGER = logging.getLogger(__name__)
+DATE_TIME_TYPES = ['date', 'timestamp', 'datetime']
+MAX_BAD_RECORDS = 0
+FIELD_DELIMITER = '\t'
+JINJA_ENV = jinja2.Environment()
+SELECT_TPL = JINJA_ENV.from_string("""
+    SELECT 
+    {% for field in fields %}
+        {% if field.field_type == 'date' %}
+            PARSE_DATE('%Y%m%d', {{ field['name'] }})
+        {% else %}
+            {{ field.name }}
+        {% endif %}
+        AS {{ field.name }} {% if loop.nextitem is defined %} , {% endif %}
+    {% endfor %}
+    FROM `{{ project_id }}.{{ dataset_id }}.{{ table }}`
+""")
+
+
+def wait_jobs(jobs: Iterable[Union[QueryJob, LoadJob]]):
+    """
+    Run multiple jobs to completion
+
+    :param jobs: jobs to run
+    :return: the completed jobs
+
+    :raises RuntimeError if a LoadJob completes with errors
+    :raises google.cloud.exceptions.GoogleCloudError if a QueryJob fails
+    """
+    _jobs = []
+    for job in jobs:
+        _job = job.result()
+        _jobs.append(_job)
+        if hasattr(_job, 'errors') and _job.errors:
+            LOGGER.error(f"Error running job {_job.job_id}: {_job.errors}")
+            raise RuntimeError(
+                f"Error running job {_job.job_id}: {_job.errors}")
+    return _jobs
+
+
+def safe_schema_for(table: str) -> List[SchemaField]:
+    """
+    Get schema fields whose date[time] fields are converted to strings so load will work
+
+    :param table: name of the table
+    :return: a list of SchemaField objects
+    """
+    return [
+        SchemaField(
+            f.name,
+            'string' if f.field_type in DATE_TIME_TYPES else f.field_type,
+            f.mode, f.description) for f in bq.get_table_schema(table)
+    ]
+
+
+def _filename_to_table_name(filename: str) -> str:
+    return filename.replace('.csv', '').lower()
+
+
+def load_stage(dst_dataset: Dataset, bq_client: Client,
+               bucket_name: str) -> List[LoadJob]:
+    """
+    Stage files from a bucket to a dataset
+
+    :param dst_dataset: reference to destination dataset object
+    :param bq_client: a BigQuery client object
+    :param bucket_name: the location in GCS containing the vocabulary files
+    :return: list of completed load jobs
+    """
+    gcs_client = storage.Client()
+    blobs = list(gcs_client.list_blobs(bucket_name))
```

Comment:
Just wondering what if the provided `bucket_name` doesn't exist, is it okay to let it fail? The error message I got was something like "you don't have access to this bucket"

Reply:
I get 
```
404 GET https://storage.googleapis.com/storage/v1/b/some-bucket-that-does-not-exist/o?projection=noAcl: Not Found
```
which I think is the appropriate way to fail